trieste/acquisition/function.py

```diff
@@ -121,41 +104,45 @@ def prepare_acquisition_function(
         """
         :param dataset: The data from the observer. Must be populated.
         :param model: The model over the specified ``dataset``.
-        :return: The expected improvement function.
+        :return: The expected improvement function. This function will raise
+            :exc:`ValueError` or :exc:`~tf.errors.InvalidArgumentError` if used with a batch size
+            greater than one.
         :raise ValueError: If ``dataset`` is empty.
         """
         if len(dataset.query_points) == 0:
             raise ValueError("Dataset must be populated.")
         mean, _ = model.predict(dataset.query_points)
         eta = tf.reduce_min(mean, axis=0)
-        return lambda at: self._acquisition_function(model, eta, at)
+        return expected_improvement(model, eta)
 
-    @staticmethod
-    def _acquisition_function(
-        model: ProbabilisticModel, eta: TensorType, at: TensorType
-    ) -> TensorType:
-        return expected_improvement(model, eta, at)
 
-
-def expected_improvement(model: ProbabilisticModel, eta: TensorType, at: TensorType) -> TensorType:
+def expected_improvement(model: ProbabilisticModel, eta: TensorType) -> AcquisitionFunction:
     r"""
-    The Expected Improvement (EI) acquisition function for single-objective global optimization.
-    Return the expectation of the improvement at ``at`` over the current "best" observation ``eta``,
-    where an improvement moves towards the objective function's minimum, and the expectation is
-    calculated with respect to the ``model`` posterior. For model posterior :math:`f`, this is
+    Return the Expected Improvement (EI) acquisition function for single-objective global
+    optimization. Improvement is with respect to the current "best" observation ``eta``, where an
+    improvement moves towards the objective function's minimum, and the expectation is calculated
+    with respect to the ``model`` posterior. For model posterior :math:`f`, this is
 
     .. math:: x \mapsto \mathbb E \left[ \max (\eta - f(x), 0) \right]
 
     This function was introduced by Mockus et al, 1975. See :cite:`Jones:1998` for details.
 
     :param model: The model of the objective function.
     :param eta: The "best" observation.
-    :param at: The points for which to calculate the expected improvement.
-    :return: The expected improvement at ``at``.
+    :return: The expected improvement acquisition function. This function will raise
+        :exc:`ValueError` if used with a batch size greater than one.
```

Comment:
```suggestion
        :exc:`ValueError`  or :exc:`~tf.errors.InvalidArgumentError` if used with a batch size greater than one.
```
`assert_shapes` can raise either afaik (depending on whether the function's compiled)

Reply:
similar in other places

Reply:
Adding them :)